packages/grafana-ui/src/components/Forms/Checkbox.tsx

```diff
@@ -42,112 +44,141 @@ export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
   }
 );
 
-export const getCheckboxStyles = stylesFactory((theme: GrafanaTheme2) => {
-  const labelStyles = getLabelStyles(theme);
-  const checkboxSize = 2;
-  const labelPadding = 1;
-
-  return {
-    wrapper: css`
-      position: relative;
-      vertical-align: middle;
-      font-size: 0;
-    `,
-    input: css`
-      position: absolute;
-      z-index: 1;
-      top: 0;
-      left: 0;
-      width: 100% !important; // global styles unset this
-      height: 100%;
-      opacity: 0;
-
-      &:focus + span,
-      &:focus-visible + span {
-        ${getFocusStyles(theme)}
-      }
-
-      &:focus:not(:focus-visible) + span {
-        ${getMouseFocusStyles(theme)}
-      }
-
-      /**
+export function getCheckboxStyles(size?: ComponentSize) {
+  return stylesFactory((theme: GrafanaTheme2) => {
```

Comment:
this will return a  new function every render, which stoåps useStyles2 from working, and stylesFactory (they both provide memoisation)